.github/workflows/builds.yml

```diff
@@ -95,6 +100,72 @@ jobs:
           include: LICENSE,README.md
           token: ${{ secrets.GITHUB_TOKEN }}
 
+  npm-publish:
+    needs: binary-builds
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target: aarch64-unknown-linux-musl
+            os: linux
+            arch: arm64
+          - target: x86_64-unknown-linux-musl
+            os: linux
+            arch: x64
+          - target: aarch64-apple-darwin
+            os: darwin
+            arch: arm64
+          - target: x86_64-apple-darwin
+            os: darwin
+            arch: x64
+          - target: x86_64-pc-windows-msvc
+            os: windows
+            arch: x64
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install cargo get
+        run: cargo install cargo-get
+      - name: Prepare os/arch packages
+        shell: bash
+        env:
+          node_os: ${{ matrix.os }}
+          node_arch: ${{ matrix.arch }}
+          node_pkg: homestar-${{ matrix.os }}-${{ matrix.arch }}
+        run: |
+          export node_version=$(cargo get workspace.package.version)
+          echo "node_pkg=${node_pkg}" >> "$GITHUB_ENV"
+          cd homestar-runtime/npm
+          mkdir -p "${node_pkg}/bin"
+          envsubst < package.json.tmpl > "${node_pkg}/package.json"
+      - name: Download build artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ matrix.target }}
+          path: "homestar-runtime/npm/${{ env.node_pkg }}/bin"
+      - run: echo ${{  github.event_name }}
+      - name: Publish production
+        if: github.event_name == 'release' && github.event.action == 'published'
+        run: |
+          cd "homestar-runtime/npm/${{ env.node_pkg }}"
+          pnpm -r publish --access=public
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+      - name: Publish RC
+        if: github.event_name == 'workflow_dispatch' || github.event_name == 'pull_request'
```

Comment:
did we want to do this on PR or merge? Now, post-workshop, we'll be merging consistently into `main`, as PR's could still have broken code?

Reply:
just testing for now, its will be just `workflow_dispatch` for RC and the production one

Reply:
Ok, makes sense.